packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -680,6 +680,8 @@ export class Function extends FunctionBase {
     if (props.vpc) {
       // Policy that will have ENI creation permissions
       managedPolicies.push(iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaVPCAccessExecutionRole'));
+    } else if (props.vpcSubnets) {
```

Comment:
I think I'm fine with this change given that it does not cause any changes to the deployed infra.

One nit - can you put this check in the `configureVpc` method? It feels like it should go there since that is where the other checks are.